tensorflow/core/kernels/adjust_hsv_op_common.h

```diff
@@ -0,0 +1,44 @@
+/* Copyright 2016 The TensorFlow Authors. All Rights Reserved.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+#include "tensorflow/core/framework/op_kernel.h"
+#include "tensorflow/core/framework/tensor.h"
+
+#ifndef TENSORFLOW_CORE_KERNELS_ADJUST_HSV_OP_COMMON_H_
+#define TENSORFLOW_CORE_KERNELS_ADJUST_HSV_OP_COMMON_H_
+
+namespace tensorflow {
+
+class AdjustHueOpBase : public OpKernel {
+
+protected:
+
+AdjustHueOpBase(OpKernelConstruction* context) : OpKernel(context) {}
+
+struct ComputeOptions {
+  const Tensor* input;
+  const Tensor* delta;
+  Tensor* output;
+  int64 channel_count;
+};
+
+virtual void DoCompute(OpKernelContext* context,
+                       const ComputeOptions& options) = 0;
+
+virtual void Compute(OpKernelContext* context);
+};
+
+template <class Device>
+class AdjustHueOp {};
```

Comment:
@zheng-xq I think it will be possible to remove it from this file when the op part of the GPU impl gets moved to the same compilation unit as the CPU code, and the GPU file only has a functor, as you pointed out below.